### DISCUSSION TITLE
Integration of the modernized theming for the Eclipse UI

### DISCUSSION THREAD

**Discussion by HannesWell** [eclipse-platform/eclipse.platform.ui#2199] [Polls]
In the past month a [modernized light and dark theme](https://github.com/eclipse-platform/eclipse.platform.ui/issues/2114) has been integrated for Eclipse. And while it has already been improved since then to fix critical problems, more fine-tuning is still pending. Furthermore Plug-ins from other projects in the Eclipse simultaneous release or third-party Plug-ins maybe need more time to adapt to the new theming, especially in edge cases.

In order to mitigate these risks the old'/previous light-theme could at least be preserved in some form. Preserving the previous dark-theme on the other hand would be a lot of work and is therefore considered not feasible. Preserving the light-theme would also allow those that simply like the old light-theme better to continue to use it. However, if that is done, it is to be discussed for how long it is preserved, especially because it will not be maintained by the providers of the modernized light-theme.

We would like to know the opinion of the broader community on this topic.
This poll closes at Tuesday 20th of August 2024 23:59UTC and is non-binding.

**Comment by BeckerWdf** [comment]
> However, if that is done, it is to be discussed for how long it is preserved, especially because it will not be maintained by the providers of the modernized light-theme.

We must also decide how to react once the "old light theme" is removed. How would installations / workspaces that have this removed theme in their settings react?

**Comment by BeckerWdf** [comment]
Thank you for your valuable input in the poll, in the pull requests and issues. 🙏

It’s great to see that the majority of you thinks that the theme changes are valuable and that you would prefer them to be the default for the 2024-09 release. However we haven given this some more thoughts and think that there are still some issues and decisions to be made. We don’t want to jeopardise the image of the Eclipse Platform and IDE by pushing these changes too early while these point are not yet addressed. 

Our plan is to:
- [x] go for option 3 (provide change in a new Theme that is not the default) for the 2024-09 release
- [x] add an entry in the platform “[New & Noteworthy](https://github.com/eclipse-platform/www.eclipse.org-eclipse/pull/200)” document about this new theme 
- [ ] make the new theme the default again early in the 2024-12 development cycle
- [ ] clarify remaining issues in the 2024-12 development cycle
- [ ] release the theme changes as the the new default (option 1) for the 2024-12 release 
- [ ] decide wether we want to keep the “old” light theme as an additional option e.g. as “Light (Previous)” (option 2)

So thanks again for the great teamwork and the fruitful discussions 🙏  . Let’s keep this up for the future! 👍

**Reply by mickaelistria** [reply]
FWIW, we've faced similar issues multiple times in the past with themes. Here is a bit of my conclusion after similar past experience with themes (last one was ~4 years ago IIRC).
Allowing an opt-(in|out) for disruptive and subjective changes such as themes is most often necessary. So this means that in any case, the "old" theme should stay until a very large majority (~80%) of the audience seems to prefer it over the new one. On that aspect, I would recommend small incremental changes in existing themes to be introduced sporadically rather than designing a whole new thing. Going step by step facilitates adoption.
Then about whether to make the new feature default or not, it's really a matter of risk perception. One cannot target 100% satisfaction with themes, those are too subjective and even so the palette and lines "trends" don't last very long. If a trendy competitor emerge next week, then it would set a new standard and make a 1-week old them in Eclipse IDE feel obsolete... So the question as to when to make it default is "whenever the majority of the community prefers it over current one, and believes the larger audience will also prefer it". It can be in next release, in the one after, in the one after, on in 5 years, or even never. There is no general good answer here.
Good luck!
(I've disabled theme engine soon after my last attempt to improve themes, so I won't vote or voice an opinion on the current proposal as I have not tested it nor do I plan to)

**Reply by vogella** [reply]
AFAIK option 3 will break the new themes, as existing extensions like syntax highlighting in JDT or dark theme extensions in Git are based on the theme ID. If you add new themes with new IDs these additional settings will not be applied

**Reply by jukzi** [reply]
> syntax highlighting in JDT or dark theme extensions in Git are based on the theme ID

That should be changed. It would be good to have the theme have an attribute indicating light or dark and a public API to ask isDarkTheme()

**Comment by BeckerWdf** [comment]
See: https://github.com/eclipse-platform/eclipse.platform.ui/pull/2206
